rust/connlib/clients/shared/src/control.rs

```diff
@@ -301,6 +275,38 @@ impl<CB: Callbacks + 'static> ControlPlane<CB> {
             .send(EgressMessages::CreateLogSink {})
             .await;
     }
+
+    pub async fn handle_tunnel_event(&mut self, event: firezone_tunnel::Event) {
+        match event {
+            firezone_tunnel::Event::SignalIceCandidate { conn_id, candidate } => {
+                let Some(gateway_id) = conn_id.into_gateway_id() else {
+                    return;
+                };
+                let ice_candidate = match candidate.to_json() {
+                    Ok(ice_candidate) => ice_candidate,
+                    Err(e) => {
+                        tracing::warn!(
+                            "Failed to serialize ICE candidate to JSON: {:#}",
+                            anyhow::Error::new(e)
+                        );
+                        return;
+                    }
+                };
+
+                // TODO: How to handle this error?
```

Comment:
We can just log the error.

If the connection can't succeed due to this, the connection will be re-attempted by an upper layer.